docs/community/release-checklist.md

```diff
@@ -0,0 +1,340 @@
+# Release Checklist
+
+> If your experience deviates from this document, please document the changes to keep it up-to-date.
+
+This guide will guide you (pun intended) on your journey to release a new version of Krustlet.
+
+All releases will be of the form `vX.Y.Z` where `X` is the major version number, `Y` is the minor
+version number and `Z` is the patch release number. This project strictly follows [semantic
+versioning](https://semver.org/) so following this step is critical.
+
+It is important to note that this document assumes that the git remote in your repository that
+corresponds to <https://github.com/deislabs/krustlet> is named `upstream`. If yours is not (for
+example, if you've chosen to name it `origin` or something similar instead), be sure to adjust the
+listed snippets for your local environment accordingly.
+
+If you are not sure what your upstream remote is named, use a command like `git remote -v` to find
+out.
+
+If you don't have an upstream remote, you can add one easily using something like:
```

Comment:
In this case, I don't think it is "something like," I think this should be the command they run if they don't have an `upstream`